src/plugins/kibana_react/public/overlays/mount_for_component.tsx

```diff
@@ -0,0 +1,35 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import React from 'react';
+import ReactDOM from 'react-dom';
+import { I18nProvider } from '@kbn/i18n/react';
+import { MountPoint } from 'kibana/public';
+
+/**
+ * Mount converter for react components.
+ *
+ * @param component to get a mount for
+ */
+export const mountForComponent = (component: React.ReactElement): MountPoint => (
+  element: HTMLElement
+) => {
+  ReactDOM.render(<I18nProvider>{component}</I18nProvider>, element);
+  return () => ReactDOM.unmountComponentAtNode(element);
+};
```

Comment:
Not sure about the name. maybe `reactMount` is better (and shorter)? WDYT?

Reply:
`reactMount`, `reactToMountPoint`, `elementToMountPoint`, `renderMountPoint`, `renderToMountPoint`.

Maybe we can change the type of `component` to `React.ReactNode`?

Reply:
good point, `ReactNode` seems more appropriate.